File: lingting-spring-elasticsearch/src/test/java/live/lingting/spring/elasticsearch/SwitchElasticsearchDataPermissionHandler.java
package live.lingting.spring.elasticsearch;

import live.lingting.framework.elasticsearch.datascope.DefaultElasticsearchDataPermissionHandler;
import live.lingting.framework.elasticsearch.datascope.ElasticsearchDataScope;
import org.springframework.stereotype.Component;

import java.util.List;

/**
 * @author lingting 2024-03-08 18:06
 */
@Component
public class SwitchElasticsearchDataPermissionHandler extends DefaultElasticsearchDataPermissionHandler {

	/**
	 * true 则不忽略权限控制
	 */
	private boolean enabled = false;

	public SwitchElasticsearchDataPermissionHandler(List<ElasticsearchDataScope> scopes) {
		super(scopes);
	}

	@Override
	public boolean ignorePermissionControl(String index) {
		return !enabled;
	}

	public void enable() {
		this.enabled = true;
	}

	public void disable() {
		this.enabled = false;
	}

}

File: lingting-spring-elasticsearch/src/test/kotlin/live/lingting/spring/elasticsearch/SwitchElasticsearchDataPermissionHandler.kt
package live.lingting.spring.elasticsearch

import live.lingting.framework.elasticsearch.datascope.DefaultElasticsearchDataPermissionHandler
import live.lingting.framework.elasticsearch.datascope.ElasticsearchDataScope
import org.springframework.stereotype.Component

/**
 * @author lingting 2024-03-08 18:06
 */
@Component
class SwitchElasticsearchDataPermissionHandler(scopes: MutableList<ElasticsearchDataScope>) : DefaultElasticsearchDataPermissionHandler(scopes) {
    /**
     * true 则不忽略权限控制
     */
    private var enabled = false

    override fun ignorePermissionControl(index: String): Boolean {
        return !enabled
    }

    fun enable() {
        this.enabled = true
    }

    fun disable() {
        this.enabled = false
    }
}
